### PR TITLE
Simplify model computation threading

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
A lot of this code existed before we even had async/await, and back when roslyn owned the entire completion stack.

it makes little sense now that:

1. we are using ConfigureAwait(false) everywhere anyways  (so we're using the theradpool all over the place)
2. we're passing TaskScheduler.Default every place we construct this type.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide @jasonmalinowski ptal.

File: src/EditorFeatures/Core.Wpf/SignatureHelp/Controller.Session_ComputeModel.cs [hunk 1]
```diff
@@ -54,7 +53,6 @@ private async Task<Model> ComputeModelInBackgroundAsync(
                 {
                     using (Logger.LogBlock(FunctionId.SignatureHelp_ModelComputation_ComputeModelInBackground, cancellationToken))
                     {
-                        this.Computation.ThreadingContext.ThrowIfNotOnBackgroundThread();
                         cancellationToken.ThrowIfCancellationRequested();
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/SignatureHelp/Controller.Session_ComputeModel.cs:56]:
```suggestion
                        this.Computation.ThreadingContext.ThrowIfNotOnBackgroundThread();
                        cancellationToken.ThrowIfCancellationRequested();
```

File: src/EditorFeatures/Core/IntelliSense/ModelComputation.cs [hunk 2]
```diff
@@ -140,9 +122,8 @@ public void ChainTaskAndNotifyControllerWhenFinished(
         var asyncToken = _controller.BeginAsyncOperation();
 
         // Create the task that will actually run the transformation step.
-        var nextTask = _lastTask.SafeContinueWithFromAsync(
-            t => transformModelAsync(t.Result, _stopCancellationToken),
-            _stopCancellationToken, TaskContinuationOptions.OnlyOnRanToCompletion, _taskScheduler);
+        var nextTask = TransformModelAsync(_lastTask);// transformModelAsync(_lastTask, _stopCancellationToken);
```
Comment (JoeRobich) [inline src/EditorFeatures/Core/IntelliSense/ModelComputation.cs:125]:
```suggestion
        var nextTask = TransformModelAsync(_lastTask);
```

**Review by JoeRobich — APPROVED** [review]

File: src/EditorFeatures/Core/IntelliSense/ModelComputation.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/IntelliSense/ModelComputation.cs:125]: oops.  will remove in followup!

**Pull request merged**